cachito/workers/pkg_managers/pip.py

```diff
@@ -2175,3 +2181,60 @@ def get_raw_component_name(requirement):
         raw_component_name = f"{repo_name}/{filename}"
 
     return raw_component_name
+
+
+def _iter_zip_file(file_path: Path):
+    with zipfile.ZipFile(file_path, "r") as zf:
+        return zf.namelist()
+
+
+def _iter_tar_file(file_path: Path):
+    with tarfile.open(file_path, "r") as tar:
+        for member in tar:
+            yield member.name
+
+
+def is_pkg_info_dir(path):
+    """Simply check whether a path represents the PKG_INFO directory.
+
+    Generally, it is in the format for example: pkg-1.0/PKG_INFO
+
+    :param str path: a path.
+    :return: True if it is, otherwise False is returned.
+    :rtype: bool
+    """
+    parts = os.path.split(path)
+    return len(parts) == 2 and parts[1] == "PKG-INFO"
+
+
+def check_metadata_in_sdist(sdist_path: Path):
+    """Check if a downloaded sdist package has metadata.
+
+    :param sdist_path: the path of a sdist package file.
+    :type sdist_path: pathlib.Path
+    :raise ValidationError: if the sdist cannot be validated.
+    """
+    if str(sdist_path).endswith(ZIP_FILE_EXT):
```

Comment:
This could be `sdist_path.name.endswith`, same for the lines below

Reply:
Fixed